react/components/enhancedCommerce.ts

```diff
@@ -0,0 +1,42 @@
+/** Product viewed event
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#measuring-actvities
+ */
+export const productDetail = (product: any) => {
+  ga('ec:addProduct', {
+    brand: product.brand,
+    category: product.categoryId,
+    id: product.productId,
+    name: product.productName,
+    variant: product.variant,
+  })
+  ga('ec:setAction', 'detail')
+  ga('send', 'event', 'Ecommerce', 'Detail', { nonInteraction: 1 })
```

Comment:
Maybe it would be better to have the event _category_ should be Ecommerce -> Product and the _action_ Detail -> View.

Reply:
I Liked